src/main/scala/mesosphere/marathon/core/task/jobs/impl/OverdueInstancesActor.scala

```diff
@@ -65,7 +65,7 @@ private[jobs] object OverdueInstancesActor {
 
       def launchedAndExpired(task: Task): Boolean = {
         task.status.condition match {
-          case Condition.Created | Condition.Starting if task.status.stagedAt < unconfirmedExpire =>
+          case Condition.Provisioned | Condition.Starting if task.status.stagedAt < unconfirmedExpire =>
```

Comment:
Side note: I've seen this part before and wondered: Why `Created/Provisioned & Starting`?

According to [the Mesos docs](https://github.com/apache/mesos/blob/004fb5fa27c2992b11a2fa51a8ec5a3f3de404db/docs/high-availability-framework-guide.md#the-life-cycle-of-a-task), It's `Staging` -> `Starting`. So it should be either

```
case Provisioned | Staging => ...
case Starting => ...
```
or
```
case Provisioned => ...
case Staging | Starting => ...
```

but this combination is weird.

Reply:
that's a great point, I see that it dates back a loong time. Maybe we should create JIRA and change that? I read the docs and even from our docs it does not make sense that unconfirmed timeout is applied on starting state

Reply:
sgtm – https://jira.mesosphere.com/browse/MARATHON-8482 😎

Reply:
`task_launch_confirm_timeout` is defined as `Time, in milliseconds, to wait for a task to enter the TASK_STAGING state before killing it` so it's the 2) (`case Provisioned => `).

P.S. Wouldn't this have been *much faster* to change this line, instead of creating another JIRA? 🙈